### ISSUE TITLE
Design Intent - Accessibility Feedback - Unified Appointment Experience, Appointments, Community Care Direct Scheduling

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#81745]
## Next Steps for the VFS team

- [x] **Assign** this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] **Comment** on this ticket:
  - If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members`.
- [x] **Close the ticket** when all feedback has been addressed.

## Thoughts/questions

- I really appreciate your patience in answering all of our questions! This one is tricky and you've put a lot of thought into it.

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform Governance reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- [ ] **Must:** I'm good with using buttons as a single-click selection of an appointment slot, rather than using radio buttons. But for screen reader users tabbing to that button, we need to make sure all of the relevant appointment information is associated with the button -- date, time, provider name.
   
   A few possible approaches to explore as you move toward implementation (definitely involve someone from CAIA):
   - I *think* you can get away with this with something like:
      ```
      <fieldset>
         <legend>Next available time with [provider name]</legend>
         <button>Monday, April 29, 2024 at 2:00PM</button>
      </fieldset>
      ```
      Or, if not doing the Next Available option, something like:
      ```
      <fieldset>
         <legend>Available appointments with [provider name] on [date]</legend>
         <button>2:00PM</button>
         <button>2:45PM</button>
         ...
      </fieldset>
      ```
      The `fieldset`/`legend` pattern is typically used with more traditional form inputs, but I think it should work for all screen readers here... but that's something to test and verify.
   - If `fieldset`/`legend` doesn't work, you could use `aria-describedby`:
      ```
      <p id="some-id">Next available time with [provider name]</p>
      <button>Monday, April 29, 2024 at 2:00PM</button>
      ```
      The downside there is (1) it flips the order in which text is read (the `describedby` will come after the button label) which is a little awkward, and (2) `aria-describedby` is not always announced with the same priority depending on screen reader users' verbosity settings.
   - One more thing to explore: Some kind of button variant that has all of the information inside the button, with text styled in some way where it doesn't look like a giant button. That's going to be your surest bet for accessibility, but also will require some design exploration and might make Matt Dingee raise his eyebrows! 😄 
- [ ] **Should:** Of the ideas presented to us, I like the Next Available option best because it seems like the best option for cognitive load. If that tests poorly, I definitely prefer the Multiple Dates option to the Day Toggle option.
- [ ] **Consider:** Thinking about some of the user flow discussions we had at the meeting... You're currently combining two tasks into a single screen -- selecting a provider and selecting a date/time. In the Next Available flow (eg. Frame 471), you streamline that by kind-of-sort-of giving Veterans a screen for selecting a provider, where the button to select the provider gives them a default selection of that provider's next available time. 
   
   Since it sounds like your research says selecting date/time is the highest priority for most Veterans, I wonder what it would be like if you flipped it. Give the users a list of available dates/times along the lines of:
      
      Choose an appointment
      - Monday, April 29, 2024 at 2:00PM with Dr. X
         - More information about Dr. X
      - Monday, April 29, 2024 at 2:45 with Dr. Y
         - More information about Dr. Y
      ... etc 
      
   I don't know if that would be better! But it might be worth looking at.
- [ ] **Consider:** Another thought on breaking up the user flow a little, possibly in combination with the other "Consider" item above -- What would it look like if "Review your approved referral..." contained two links -- one for choosing provider and then date/time, the other for just browsing available appointments regardless of provider? In the meeting we were trying to get into the headspace of Veterans using this feature and trying to think how they might approach scheduling, but maybe there's value in just letting Veterans decide which mental model they want to follow.

## Governance team actions
- [x] Format feedback as individual tasks (check boxes)
- [x] Assign this ticket to the VFS team member that opened the Slack request
- [x] Add the VFS team product label
- [x] Add the VFS team the feature label (if applicable)
- [x] Add the touchpoint labels
- [x] Add the practice area labels
- [x] Add the Collaboration Cycle initiative milestone
- [x] Add the Collaboration Cycle initiative epic

**Comment by benbrasso-agile6** [comment]
Thanks, Brian!

**Comment by cgednalske** [comment]
We brought our updated approach to office hours on 5/8 ([Figma mockups](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=629%3A31344&t=mn1Pq4pRTye1E0jS-1)), which should adequately address all accessibility feedback.

> **Must:** I'm good with using buttons as a single-click selection of an appointment slot, rather than using radio buttons. But for screen reader users tabbing to that button, we need to make sure all of the relevant appointment information is associated with the button -- date, time, provider name.
  -  Our updated approach separates out provider selection and date selection, and we'll be using the existing radio button approach from Appointments self-scheduling for our [time/date selection page](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=633%3A31926&t=mn1Pq4pRTye1E0jS-1). With these changes, we think this feedback is no longer relevant, but we'll make sure to reference this ticket and check in with a11y when this is coded. 

> **Should:** Of the ideas presented to us, I like the Next Available option best because it seems like the best option for cognitive load. If that tests poorly, I definitely prefer the Multiple Dates option to the Day Toggle option.
- The [updated provider list](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=629%3A35128&t=mn1Pq4pRTye1E0jS-1) includes each provider's next available appointment, but we removed the interaction to simplify the process while still providing this information.


> **Consider:** Thinking about some of the user flow discussions we had at the meeting... You're currently combining two tasks into a single screen -- selecting a provider and selecting a date/time. In the Next Available flow (eg. Frame 471), you streamline that by kind-of-sort-of giving Veterans a screen for selecting a provider, where the button to select the provider gives them a default selection of that provider's next available time.
  - At office hours on 5/8, we brought an updated approach that separates out the provider selection and date/time selection steps, and lists next available appointment in the provider list information. ([Figma mockups](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=629%3A31344&t=mn1Pq4pRTye1E0jS-1)) 

> **Consider:** Another thought on breaking up the user flow a little, possibly in combination with the other "Consider" item above -- What would it look like if "Review your approved referral..." contained two links -- one for choosing provider and then date/time, the other for just browsing available appointments regardless of provider? In the meeting we were trying to get into the headspace of Veterans using this feature and trying to think how they might approach scheduling, but maybe there's value in just letting Veterans decide which mental model they want to follow.

  - We brought this option to office hours as well, but the group agreed that the combined approach ([provider list with the option to filter/open a calendar and choose a specific date](https://www.figma.com/design/DsRXEFiYLCFnY5nBkp9Dc4/CC-Referral-%7C-Appointments-FE?node-id=629%3A35128&t=mn1Pq4pRTye1E0jS-1)) better addresses Veterans' scheduling needs, and allows them flexibility in how to schedule without forcing a decision early on.
